### ISSUE TITLE
Custom components are updating into incorrect folder

### ISSUE THREAD

**Issue by jamespreedy** [hacs/integration#2437]
### System Health details

System Health

version: core-2021.12.8
installation_type: Home Assistant OS
dev: false
hassio: true
docker: true
user: root
virtualenv: false
python_version: 3.9.7
os_name: Linux
os_version: 5.10.88
arch: x86_64
timezone: Europe/London

GitHub API: ok
Github API Calls Remaining: 4928
Installed Version: 1.19.1
Stage: running
Available Repositories: 996
Downloaded Repositories: 13

logged_in: true
subscription_expiration: 1 February 2022, 12:00 am
relayer_connected: true
remote_enabled: true
remote_connected: true
alexa_enabled: false
google_enabled: false
remote_server: eu-west-2-0.ui.nabu.casa
can_reach_cert_server: ok
can_reach_cloud_auth: ok
can_reach_cloud: failed to load: timeout

host_os: Home Assistant OS 7.1
update_channel: stable
supervisor_version: supervisor-2021.12.2
docker_version: 20.10.9
disk_total: 222.8 GB
disk_used: 19.5 GB
healthy: true
supported: true
board: generic-x86-64
supervisor_api: ok
version_api: ok
installed_addons: File editor (5.3.3), SSH & Web Terminal (10.0.1), Check Home Assistant configuration (3.9.0), SQLite Web (3.2.0), Duck DNS (1.14.0), Z-Wave JS to MQTT (0.32.2)

dashboards: 1
resources: 0
views: 11
mode: storage

### Checklist

- [X] I'm running the newest version of HACS <https://github.com/hacs/integration/releases/latest>
- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] I have read <https://hacs.xyz/docs/issues>
- [X] This issue is related to the backend (integration part) of HACS.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).

### Describe the issue

Hiya, thanks in advance for your help.
I'm unsure if this is an issue with HACS, or with the custom components themselves, but it seems like a coincidence.

The last two custom components that have had HACS updates, have updated fine.
However, when rebooting Home Assistant, I get an error saying that the component could not be loaded.

When looking in my custom component folder, the updated component was installed in the wrong location.
For example ...

/config/custom_components/heatmiserneo/manifest.json
... was actually in ...
/config/custom_components/heatmiserneo/heatmiserneo/manifest.json

/config/custom_components/unifiprotect/manifest.json
... was actually in ...
/config_custom_components/unifiprotect/unifiprotect/manifest.json

It was almost as if an additional folder was created that the component was moved into.
Home Assistant wasn't able to see them as they were in the wrong location.

Once I removed that duplicate folder and moved all the files into the correct location, those components started up fine when I rebooted Home Assistant.

This happened on the last two components I updated, so I've enabled debug for HACS now, and will include logs when my next component needs updating in case I get three in a row!

Thanks for your help, I'll keep this updated.

Again, its perhaps co-incidental, but I only had this issue after installing HACS 1.19.0.
After updating to 1.19.1, I still had the same issue.

### Reproduction steps

1. Install HACS component update
2. Reboot Home Assistant
3. Check logs to see if updated custom_component has loaded successfully
4. If its not loaded successfully then check folder structure within /config/custom_components/ for duplicate folders
5. Resolve folder structure manually and reboot Home Assistant

### Debug logs

```text
I've enabled debug logging for HACS and will include these when my next component needs updating. I appreciate this isn't super helpful, but I wanted to give you a heads-up just in case this wasn't a co-incidence. Thanks for your help!
```

**Comment by ludeeus** [comment]
This was fixed to 1.19.1
Enable debug for HACS, restart and redownload one of the integrations you have issues with, and post the logs for that.

**Comment by modem** [comment]
I just had the same issue with 1 integration updated yesterday and another one today using the latest version of HACS 1.91.1.
I re-downloaded them both today and they are now ok, but seems the initial issue with the update is still there, otherwise I would not had issues with the upgrade from today.
I have no logs as I've just found this issue after I re-downloaded the integrations.
